### PR TITLE
Integrate Android side of React Native Library

### PR DESCRIPTION
**Pull request by gmale**
The corresponding app changes to add functionality for [this PR](https://github.com/EdgeApp/react-native-zcash/pull/4)

### CODE REVIEW

**Comment by gmale** [comment]
Some screenshots I took while working on this.

File: android/app/src/main/java/com/zcashdemoapp/App.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.zcashdemoapp
```
Comment (paullinator) [inline android/app/src/main/java/com/zcashdemoapp/App.kt:1]: is it necessary to migrate the parent application to Kotlin? Most React Native apps, if not all, have the main app as Java. It would be nice to be able to use react-native-zcash without having to modify the parent app, outside of including the package.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/com/zcashdemoapp/App.kt (hunk 1 above)
Comment (gmale) [reply android/app/src/main/java/com/zcashdemoapp/App.kt:1]: Great question!
Necessary? no. Helpful? Probably . . . because Kotlin apps can work with Java libraries but not all Java apps can work with all Kotlin libraries. Moreover, you can enable Kotlin in your parent application without needing to change your Java code--both Java and Kotlin can co-exist in one project. In fact, you could do so and then measure the impact on app size and it would likely be negligable, especially if you have proguard enabled.

From what I've seen, my suspicion would be that most React Native apps are in Java simply because that's the output of the default project initializers like Expo CLI. On the other hand, most Android projects and libraries are in Kotlin. To be fair, not all of them _require_ Kotlin but Kotlin is, overwhelmingly, the direction that Android has headed.

Ultimately, it boils down to packaging. It seems that the `updot` tool pulls over the files from the library project rather than having the library compile those files directly into an artifact.

There might be a middle ground. My React native skills are non-existent but I strongly suspect it might be possible to package the `react-native-android` as a proper Android library (AAR) in such a way that React Native can use it. This would eliminate the need of using `updot` and the library could be included using gradle, directly. In the same way that I [added the android library as a gradle dependency](https://github.com/EdgeApp/react-native-zcash/pull/4/files#diff-197b190e4a3512994d2cebed8aff5479ff88e136b8cc7a4b148ec9c3945bd65aR58), the react-native-zcash could probably be packaged as an AAR and distributed to `jcenter` and pulled into android projects as a gradle dependency.

To summarize, there are probably two viable options:
1) Add Kotlin support to the base project by writing [one line of code](https://github.com/EdgeApp/zcash-demo-app/pull/1/commits/ab6e393eec685af041fe08ad921bee3b8f87b2a6#diff-9526ccfd1d1813ed49c39f8c54dbeb512607376a007d824b905bc8b4e4d202d9R84)
2) Compile the react-native-zcash library into an Android library

Comment (paullinator) [reply android/app/src/main/java/com/zcashdemoapp/App.kt:1]: Adding support for kotlin is totally fine. I was mostly curious if changing the application code itself to KT was required. If not it's probably preferable to leave the demo app in Java

Comment (paullinator) [reply android/app/src/main/java/com/zcashdemoapp/App.kt:1]: note that `updot` is only used for development purposes. The proper way to include the library would be to publish an actual version to NPM and mention that version in the parent application. Whether or not you compile and commit the compiled binaries to NPM is a matter of preference and convenience.
